raiden/network/rpc/client.py

```diff
@@ -223,10 +224,9 @@ def gaslimit(self, location='latest') -> int:
         gas_limit = self.web3.eth.getBlock(location)['gasLimit']
         return gas_limit * 8 // 10
 
-    def gas_price(self, transaction: Dict = None) -> int:
-        if transaction is None:
-            transaction = dict()
-        return int(self.web3.eth.generateGasPrice(transaction_params=transaction))
+    @cachedmethod(attrgetter('_gasprice_cache'))
+    def gas_price(self) -> int:
+        return int(self.web3.eth.generateGasPrice())
```

Comment:
Why remove the transaction argument?

Reply:
Because a dict is not hashable and we would need to use some tricks to make it cacheable.

I checked the strategies (time-based and rpc) and they do not make use of the transaction argument anyways, so it's safe to remove it. I'll add a comment to make this clearer

Reply:
The idea is that you can have decisions inside the gas price strategies that depend on the tx to be sent. E.g. if the tx uses more than X gas, use this gas price A, otherwise use gas price B. 

But we don't need that atm.

Reply:
Thanks for the explanation. @andrevmatos as @palango noted the gas price can depend on the transaction. All the more reason to report this problem upstream to web3.